tfhe/src/integer/server_key/radix_parallel/sub.rs

```diff
@@ -257,4 +260,211 @@ impl ServerKey {
+                tmp_lhs = ctxt_left.clone();
+                self.full_propagate_parallelized(&mut tmp_lhs);
+                (&tmp_lhs, ctxt_right)
+            }
+            (false, false) => {
+                tmp_lhs = ctxt_left.clone();
+                tmp_rhs = ctxt_right.clone();
+                rayon::join(
+                    || self.full_propagate_parallelized(&mut tmp_lhs),
+                    || self.full_propagate_parallelized(&mut tmp_rhs),
+                );
+                (&tmp_lhs, &tmp_rhs)
+            }
+        };
+
+        self.unchecked_unsigned_overflowing_sub_parallelized(lhs, rhs)
+    }
+
+    pub fn unchecked_unsigned_overflowing_sub_parallelized(
+        &self,
+        lhs: &RadixCiphertext,
+        rhs: &RadixCiphertext,
+    ) -> (RadixCiphertext, Ciphertext) {
+        assert!(
+            lhs.blocks.len() >= rhs.blocks.len(),
+            "Left hand side must must have a number of blocks greater or equal \
+            to the number of blocks of the right hand side: lhs {} blocks, rhs {} blocks",
+            lhs.blocks.len(),
+            rhs.blocks.len(),
+        );
+        if self.is_eligible_for_parallel_single_carry_propagation(lhs) {
+            // Here we have to use manual unchecked_sub on shortint blocks
+            // rather than calling integer's unchecked_sub as we need each subtraction
+            // to be independent from other blocks.
+            let ct = lhs
+                .blocks
+                .iter()
+                .zip(rhs.blocks.iter())
+                .map(|(lhs_block, rhs_block)| self.key.unchecked_sub(lhs_block, rhs_block))
+                .collect::<Vec<_>>();
+            let mut ct = RadixCiphertext::from(ct);
+
+            let generates_or_propagates = self.generate_init_borrow_array(&ct);
+            let (input_borrows, mut output_borrow) =
+                self.compute_borrow_propagation_parallelized_low_latency(generates_or_propagates);
+
+            ct.blocks
+                .par_iter_mut()
+                .zip(input_borrows.par_iter())
+                .for_each(|(block, input_borrow)| {
+                    // Do a true lwe subtraction, as unchecked_sub will adds a correcting term
+                    // to avoid overflow (and trashing padding bit). Here we know each
+                    // block in the ciphertext is >= 1, and that input borrow is either 0 or 1
+                    // so no overflow possible.
+                    crate::core_crypto::algorithms::lwe_ciphertext_sub_assign(
+                        &mut block.ct,
+                        &input_borrow.ct,
+                    );
+                    self.key.message_extract_assign(block);
+                });
+            assert!(ct.block_carries_are_empty());
+            // we know here that the result is a boolean value
+            // however the lut used has a degree of 2.
+            output_borrow.degree.0 = 1;
+            //assert!(output_borrow.degree.0 == 1);
```

Comment:
still required ?

Reply:
Depends on how we see things, the max value of the lut used is 2, but we know that the output carry block of the propagation steps will always be either 0 or 1.

So if I don't force the degree, we may have the problem of the block not looking like it encrypts a boolean value, when we know it in fact does

Reply:
I mean the commented assert ;)

Reply:
😅 no its not needed, I think i put it there when debugging the division that uses that overflowing_sub